homework/index.js

```diff
@@ -30,18 +31,102 @@
     return elem;
   }
 
+  function creatHeader(repositories, root) {
+    const header = createAndAppend('header', root, { class: 'header' });
+    createAndAppend('h2', header, {
+      text: 'HYF repositories',
+      class: 'nav-title',
+    });
+    const select = createAndAppend('select', header, { id: 'repo-selector' });
+    repositories.forEach((repo, index) => {
+      createAndAppend('option', select, { value: index, text: repo.name });
+    });
+    select.addEventListener('change', event => {
+      const selectedIndex = event.target.value;
+      const selectedRepository = repositories[selectedIndex];
+      while (mainContainerForRepo.firstChild) {
+        mainContainerForRepo.removeChild(mainContainerForRepo.firstChild);
+      }
+      createDescription(selectedRepository, root);
+      createContributors(selectedRepository, root);
+    });
+  }
+  const mainContainerForRepo = document.querySelector('.container');
```

Comment:
thanks for the explanation .
indeed it is outside the function because i build the document on two d
DIVs , one for the header ( which is root ) , and the other for Description , and Contributors ( which is  mainContainerForRepo ) , if this is not acceptable i will try to rebuild the document to make in one div (root) .?

Reply:
Currently, you have this structure:

```html
<body>
  <div id="root">
    <header class="header">
      <h2 class="nav-title">HYF repositories</h2>
      <select id="repo-selector">
        <option value="0">AngularJS</option>
        <option value="1">tdd-game</option>
        <!-- ... -->
        </select>
    </header>
  </div>
  <div class="container">
    <div class="left-div">
      <!-- ... -->
    </div>
    <div class="right-div">
      <!-- ... -->
    </div>
  </div>
  <script src="./index.js"></script>
</body>
```

Since you have a separate `<header>` element that contains your complete header definition, you can add your container div to `root`, instead of `<body>`. So `root` becomes the true 'root', that contains your header as well as your main container, like this:

```html
<body>
  <div id="root">
    <header class="header">
      <h2 class="nav-title">HYF repositories</h2>
      <select id="repo-selector">
        <option value="0">AngularJS</option>
        <option value="1">tdd-game</option>
        <!-- ... -->
      </select>
    </header>
    <div class="container">
      <div class="left-div"><!-- ... --></div>
      <div class="right-div"><!-- ... --></div>
    </div>
  </div>
  <script src="./index.js"></script>
</body>
```